src/path.jl

```diff
@@ -767,41 +767,37 @@ Base.write(fp::AbstractPath, x) = open(io -> write(io, x), fp, "w")
 # Default `touch` will just write an empty string to a file
 Base.touch(fp::AbstractPath) = write(fp, "")
 
-Base.tempname(::Type{<:AbstractPath}) = Path(tempname())
-tmpname() = tempname(SystemPath)
-
-Base.tempdir(::Type{<:AbstractPath}) = Path(tempdir())
-tmpdir() = tempdir(SystemPath)
-
-Base.mktemp(P::Type{<:AbstractPath}) = mktemp(tempdir(P))
-mktmp() = mktemp(SystemPath)
-
-Base.mktemp(fn::Function, P::Type{<:AbstractPath}) = mktemp(fn, tempdir(P))
-mktmp(fn::Function) = mktemp(fn, SystemPath)
-
-Base.mktempdir(P::Type{<:AbstractPath}) = mktempdir(tempdir(P))
-mktmpdir() = mktempdir(SystemPath)
-
-Base.mktempdir(fn::Function, P::Type{<:AbstractPath}) = mktempdir(fn, tempdir(P))
-mktmpdir(fn::Function) = mktempdir(fn, SystemPath)
+Base.tempname(P::Type{<:AbstractPath}; kwargs...) = tempname(tempdir(P); kwargs...)
```

Comment:
The default behaviour for `tempname` now calls `tempdir(P)`, so we require that all new path types define that method for their specific type.